tests/unit_tests/test_lib/test_printer_asyncio.py

```diff
@@ -0,0 +1,60 @@
+import contextlib
+import time
+from typing import Any, Iterator, List
+from unittest.mock import patch
+
+import pytest
+from wandb.sdk.lib import printer, printer_asyncio
+from wandb.sdk.lib.printer import _PrinterTerm
+
+
+class PrinterWrapper(_PrinterTerm):
+    """A test printer that captures text set with set_text."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._captured_text: List[str] = []
+
+    @property
+    def captured_text(self) -> List[str]:
+        return self._captured_text
+
+    @contextlib.contextmanager
+    def dynamic_text(self) -> Iterator[printer.DynamicText | None]:
+        class TestDynamicText(printer.DynamicText):
+            def __init__(self, printer: PrinterWrapper) -> None:
+                self._printer = printer
+
+            def set_text(self, text: str) -> None:
+                self._printer._captured_text.append(text)
+
+        yield TestDynamicText(self)
+
+
+def test_run_async_with_spinner():
+    test_printer = PrinterWrapper()
+
+    with patch.object(printer, "new_printer", return_value=test_printer):
+
+        def slow_func() -> int:
+            time.sleep(0.5)
+            return 42
+
+        result = printer_asyncio.run_async_with_spinner("Loading", slow_func)
+        assert result == 42
+        assert len(test_printer.captured_text) > 0
+        assert all("Loading" in text for text in test_printer.captured_text)
```

Comment:
Likewise: it's OK for the test, but a deterministic assertion is always better. Something like

```
assert test_printer.captured_text == [
    "<SPINNER 1> Loading",
    "<SPINNER 2> Loading",
    "<SPINNER 3> Loading",
]
```

That's only possible with a properly faked clock.